src/main/java/com/github/xabgesagtx/bots/TelegramBotAutoConfiguration.java

```diff
@@ -31,70 +33,84 @@
 public class TelegramBotAutoConfiguration {
 
 	private List<BotSession> sessions = new ArrayList<>();
-	
+
 	private final List<LongPollingBot> pollingBots;
-	private final List<TelegramWebhookBot> webHookBots;
+	private final List<TelegramWebhookBotService> webHookBots;
 	private final TelegramProperties properties;
+	private SetWebhook.SetWebhookBuilder webhookBuilder;
 
-	static {
-		ApiContextInitializer.init();
-	}
-	
 	@PostConstruct
-	public void start() throws TelegramApiRequestException {
+	public void start() throws TelegramApiException {
 		log.info("Starting auto config for telegram bots");
+		webhookBuilder = SetWebhook.builder();
 		TelegramBotsApi api = telegramBotsApi();
 		pollingBots.forEach(bot -> {
 			try {
 				log.info("Registering polling bot: {}", bot.getBotUsername());
 				sessions.add(api.registerBot(bot));
 			} catch (TelegramApiException e) {
 				log.error("Failed to register bot {} due to error", bot.getBotUsername(), e);
-			}			
+			}
 		});
 		webHookBots.forEach(bot -> {
 			try {
 				log.info("Registering web hook bot: {}", bot.getBotUsername());
-				api.registerBot(bot);
+				SetWebhook webhook = bot.customizeWebHook(webhookBuilder).build();
```

Comment:
The SetWebhookBuilder is mutable. This means multiple calls to customize are modifying the same builder, which could lead to unwanted effects. To avoid this you can create a new webhookBuilder for each WebHookBot you want to register

Reply:
I was previously at this point but forgotten to change it. I totally see the side effect point.
For me there are two ways: 
a) do a deep copy of the builder before passing it.
b) prepare the webhook (refactor the `telegramBotsApi` method) each time a webhook bot is instantiated. 

What do you prefer? b) is totally fine for me as we don't expect a lot of bots created and separation is much more cleare.

Reply:
I think I would also prefer b). Might be easier to understand/read